test/middleware_test.rb

```diff
@@ -81,10 +81,28 @@ def test_call_returns_403_on_invalid_nonce
   def test_call_returns_403_on_expired_timestamp
     @lti_app.config.nonce_validator = true
     @lti_app.config.time_limit      = 30
+    timestamp = (Time.now - 60*60).to_i
 
     @lti_app.stub(:valid_request?, true) do
-      env      = Rack::MockRequest.env_for('/lti/launch',
-                                           oauth_timestamp: Time.now - 60*60)
+      env = Rack::MockRequest.env_for(
+        '/lti/launch',
+        params: { oauth_timestamp: timestamp }
```

Comment:
turns out these mock requests were not being formed correctly as the options hash expected a top level params key. See the source for the method here: http://www.rubydoc.info/gems/rack/Rack%2FMockRequest.env_for